src/Overlay/GlobalOverlay/GlobalOverlay.tsx

```diff
@@ -40,9 +43,29 @@ export const GlobalOverlay: FC<GlobalOverlayProps> = ({
   });
 
   useEffect(() => {
-    if (open && children !== undefined) {
-      disableBodyScroll(children);
+    if (open && overlayRef.current !== undefined) {
+      disableBodyScroll(overlayRef.current, {
+        // allowTouchMove determines which elements to allow touchmove events for iOS https://github.com/rick-liruixin/body-scroll-lock-upgrade?tab=readme-ov-file#allowtouchmove
+        //@ts-expect-error: allowTouchMove is typed wrong: https://github.com/rick-liruixin/body-scroll-lock-upgrade/issues/21
+        allowTouchMove: (el: HTMLElement) => {
+          while (el && el !== document.body) {
+            if (el.getAttribute('body-scroll-lock-ignore') !== null) {
+              return true;
+            }
+            if (el.parentElement !== null) {
+              el = el.parentElement;
+            }
+          }
+          return false;
+        }
+      });
+    } else {
+      clearAllBodyScrollLocks();
     }
+
+    return () => {
+      clearAllBodyScrollLocks();
+    };
```

Comment:
These `clearAllBodyScrollLocks` are fixing an unrelated issue where triggering GlobalOverlay leaves `body` unscrollable even after closing the overlay

https://github.com/reaviz/rdk/assets/40581813/a671b459-9b0c-4a84-88ac-5c6a47e378dc